cpp/src/reductions/reductions.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/reduction.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include "reduction_functions.hpp"
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+// Allocate storage for a single identity scalar
```

Comment:
What is an "identity scalar"? Why is it different from any other scalar?

Reply:
it's a helper function to have common interface to create a scalar because `make_string_scalar` interface is different from `make_numeric_scalar` and `make_timestamp_scalar`.